src/lib/engines/layerCompositor/Layer.ts

```diff
@@ -75,6 +76,25 @@ export enum LayerBlendMode {
   Luminosity = 27,
 }
 
+const { Zero, One, SourceAlpha: SA, DestAlpha: DA, OneMinusSourceAlpha: InvSA, OneMinusDestAlpha: InvDA } = BlendFunc;
+const layerBlendModeBlendState: { [mode in LayerBlendMode]?: BlendState } = {
+  [LayerBlendMode.Clear]: new BlendState(Zero, Zero, Zero, Zero),
```

Comment:
I would have just done it entirely inside the shader (except for src over), but this is fine

Reply:
I had that at first, but this way we can avoid having to copy to the read buffer for trivial blends.